### ISSUE TITLE
Use binary radix in `bitflags!`

### ISSUE THREAD

**Issue by Centri3** [rust-lang/rust-clippy#11251]
### What it does

Checks for use of hexadecimal (`0x`), decimal, octal (`0o`), etc in a `bitflags!` invocation, suggesting `0b` instead.

### Advantage

- It's a bitflags type, so using binary is easier to understand as it doesn't require mental math to convert between the two (e.g., what bit is set for `0x1000`?).

### Drawbacks

- Can be worse in cases where it matches external constants, like for example: https://learn.microsoft.com/en-us/windows/win32/Memory/memory-protection-constants, winapi documentation tends to use hexadecimal instead, so I'd expect a `Protection` bitflags to use it too. So it should be `restriction` or `pedantic`.

### Example

```rust
bitflags! {
	#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
	struct Flags: u32 {
		const A = 0x1000;
	}
}
```

Could be written as:

```rust
bitflags! {
	#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
	struct Flags: u32 {
		const A = 0b1000000000000;
	}
}
```

**Comment by Centri3** [comment]
We tend not to add lints for external crates but I think `bitflags` is common enough to warrant it :)
